hosting/src/components/TogglePublishDocument.tsx

```diff
@@ -0,0 +1,37 @@
+import {Button} from "@/components/Button";
+import {useCrudTanamDocument, useTanamDocument} from "@/hooks/useTanamDocuments";
+import {serverTimestamp} from "firebase/firestore";
+import {useParams} from "next/navigation";
+
+export function TogglePublishDocument() {
+  const {documentId} = useParams<{documentId: string}>() ?? {};
+  const {data: document} = useTanamDocument(documentId);
+  const {update} = useCrudTanamDocument();
+
+  async function onTogglePublishDocument() {
+    let data = document?.toJson();
+
+    if (document?.status === "unpublished") {
+      data = {...data, status: "published", publishedAt: serverTimestamp()};
+    }
+
+    if (document?.status === "published") {
+      data = {...data, status: "unpublished", publishedAt: null};
+    }
+
+    await update(documentId, {...data});
```

Comment:
You can make this method take in a single parameter of `TanamDocumentClient` type. Do the serialization to JSON inside of the hook.

Try to always keep the full instance of the class representation for all handling inside of the application and as much as possible when passing between methods. This allows you to isolate the internal logic of _"how do we update things"_ inside of the hook.

This is better isolation of responsibilities instead of spreading out serialization logic across components.